app/Services/MenuService.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace App\Services;
+
+use App\Repo\MenuRepositoryInterface;
+use App\Models\Menu;
+use App;
+use Illuminate\Support\Collection;
+
+class MenuService extends BaseService
+{
+    /**
+     * @var \App\Repo\MenuRepositoryInterface
+     */
+    protected $menuRepository;
+
+    /**
+     *
+     * @param \App\Repo\MenuRepositoryInterface $menuRepository
+     */
+    public function __construct(MenuRepositoryInterface $menuRepository)
+    {
+        $this->menuRepository = $menuRepository;
+    }
+
+    public function find($id)
```

Comment:
validate?

Reply:
đoạn này thì validate kiểu gì Hoàng nhỉ, chẳng lẽ query ra xong rồi lại truyền id vào để query theo BaseRepository lần nữa hay sao. Tại vì chỗ này mình ko dùng được Model Binding nữa

Reply:
dùng được model binding chứ em, chỉ cần dùng ở đây là xong mà không cần quan tâm validate gì hết, bởi vì thằng model binding có validate nếu id không tồn tại rồi

Reply:
Ý em là nếu dùng model binding ở đây thì trong Service class đã lấy dc model rồi, mà pattern yêu cầu thao tác với DB phải viết trong Repository
Nếu dùng binding ở đây thì chẳng lẽ mình có đươc model rồi, sau đó lấy id của model vừa được binding truyền tiếp vào hàm `$this->menuRepository->findById($menu->id)` hả anh. Đoạn này em cũng chưa biết thế nào cho tối ưu ạ

Reply:
quy trình là em dùng model binding bên ngoài controller, lấy được model ngoài controller rồi thì khỏi cần dùng hàm này trong service

Reply:
Oki anh ạ